pinot-plugins/pinot-file-system/pinot-s3/src/main/java/org/apache/pinot/plugin/filesystem/S3PinotFS.java

```diff
@@ -93,7 +95,16 @@ public void init(PinotConfiguration config) {
         awsCredentialsProvider = DefaultCredentialsProvider.create();
       }
 
-      _s3Client = S3Client.builder().region(Region.of(region)).credentialsProvider(awsCredentialsProvider).build();
+      S3ClientBuilder s3ClientBuilder = S3Client.builder().region(Region.of(region)).credentialsProvider(awsCredentialsProvider);
```

Comment:
Just curious, does MinIO take region/awsCredentialProvider as parameters?

Reply:
I don't think MinIO cares about region.  It doesn't in the other code I'm using.

MinIO can be set up with a couple auth types.  I use accessKey and secretKey with MinIO locally.

Reply:
It might make sense then to make region optional.  I wasn't able to test this locally (given I can't build Pinot).  I think keeping it required is a safer diff.

Reply:
I think your code to not touch this should be ok, we can clean this up later.